tasks/pin_compiler.rb

```diff
@@ -0,0 +1,31 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+require_relative '../../ruby_task_helper/files/task_helper.rb'
+require 'puppet'
+require 'net/http'
+
+Puppet.initialize_settings
+
+# Turn comma separated agents into an array
+agents = JSON.parse(ARGF.read)['agent_certnames'].split(',')
+
+# Create a Net::HTTP object and set the auth to use our certificates
+http = Net::HTTP.new(uri.host, uri.port)
```

Comment:
So URI was up top for this usage of the variable.

```
==> compiler:             "msg": "The task failed with exit code 1:\n/tmp/76ce77e2-243a-474a-8e24-157f02d3be28/deploy_pe/tasks/pin_compiler.rb:12:in `<main>': undefined local variable or method `uri' for main:Object (NameError)\nDid you mean?  URI\n",
```

Maybe we move line 20 to line 10?

Reply:
Done.